documentation/modules/cruise-control/con-optimization-goals.adoc

```diff
@@ -0,0 +1,133 @@
+
+[source]
+RackAwareGoal; ReplicaCapacityGoal; DiskCapacityGoal; NetworkInboundCapacityGoal; NetworkOutboundCapacityGoal; CpuCapacityGoal; ReplicaDistributionGoal; PotentialNwOutGoal; DiskUsageDistributionGoal; NetworkInboundUsageDistributionGoal; NetworkOutboundUsageDistributionGoal; CpuUsageDistributionGoal; TopicReplicaDistributionGoal; LeaderReplicaDistributionGoal; LeaderBytesInDistributionGoal; PreferredLeaderElectionGoal
+
+To reduce complexity, we recommend that you use the inherited master optimization goals, unless you need to _completely_ exclude one or more goals from users. The priority order of the master optimization goals can be modified, if desired, in the default optimization goals (see below).
+
+Master optimization goals are controlled, if needed, in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.goals`
+
+* To accept the inherited master optimization goals, specify an empty value for the `goals` configuration option:
++
+[source=yaml]
+----
+goals: {}
+----
+
+* If you need to modify the inherited master optimization goals, specify a list of goals in the `goals` configuration option.
+
+[id="default-goals"]
+[discrete]
+=== Default optimization goals
+
+We recommend that you always set _default optimization goals_ for the Kafka cluster.
+You can include any or all of the master optimization goals in the default optimization goals.
+
+Default optimization goals are set in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.default.goals`
+
+The default optimization goals generate the _cached optimization proposal_ from the workload model (unless you set user-provided optimization goals in a `KafkaRebalance` custom resource). 
+This is updated to reflect the current state of the Kafka cluster every 15 minutes by default. 
+If no default optimization goals are specified, the cached proposal is generated using the master optimization goals.
+
+[id="user-provided-goals"]
+[discrete]
+=== User-provided optimization goals
+
+_User-provided optimization goals_ narrow down the default goals.
+You can set them, if required, in the `KafkaRebalance` custom resource for a particular optimization proposal: `KafkaRebalance.spec.goals`
+
+They are useful for generating an optimization proposal that addresses a particular scenario.
+For example, you might want to optimize leader replica distribution across the Kafka cluster without considering goals for disk capacity or disk utilization. 
+So, you create a `KafkaRebalance` custom resource containing a user-provided goal for leader replica distribution only.
+
+User-provided optimization goals must:
+
+* Include all configured hard goals, or an error will occur
+* Be a subset of the master optimization goals
+
+To exclude the configured hard goals from an optimization proposal, add the `skiphardgoalscheck: true` option to the `KafkaRebalance` custom resource.
```

Comment:
So that option will not exclude the `hard.goal`, it will just mean that CC does not complain if you miss any of the `hard goals` out of the user supplied rebalance goals. 

This should probably read something like: "If, for some reason, you want to rebalance based on a list of goals that do not include all the configured `hard.goals` then add the `skiphardgoalscheck: true`...."

Reply:
the field is `skipHardGoalCheck`